File: app/src/main/java/art/coded/wireframe/viewmodel/PlaceholderViewModel.java
package art.coded.wireframe.viewmodel;

import androidx.lifecycle.LiveData;
import androidx.lifecycle.MutableLiveData;
import androidx.lifecycle.ViewModel;

public class PlaceholderViewModel extends ViewModel {

    static final String LOG_TAG = PlaceholderViewModel.class.getSimpleName();

    private MutableLiveData<String> mText;

    public PlaceholderViewModel() {
        mText = new MutableLiveData<>();
        mText.setValue("This is placeholder fragment");
    }

    public LiveData<String> getText() {
        return mText;
    }
}

File: app/src/ktx/java/art/coded/wireframe/viewmodel/PlaceholderViewModel.kt
package art.coded.wireframe.viewmodel

import androidx.lifecycle.LiveData
import androidx.lifecycle.ViewModel
import androidx.lifecycle.MutableLiveData

class PlaceholderViewModel : ViewModel() {
    private val mText: MutableLiveData<String?>
    val text: LiveData<String?>
        get() = mText

    companion object {
        val LOG_TAG = PlaceholderViewModel::class.java.simpleName
    }

    init {
        mText = MutableLiveData()
        mText.value = "This is placeholder fragment"
    }
}